### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [projectchrono/chrono#128]
[Extra null pointer checks are not needed in functions](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null "Do I need to check for null before delete p?") like the following.
* [ChConstraintThreeGeneric](https://github.com/projectchrono/chrono/blob/54807ac75a4920c7b87a08725370b82a4cc5c960/src/chrono/solver/ChConstraintThreeGeneric.cpp#L45 "Destructor for ChConstraintThreeGeneric")
* [ChGaussPoint](https://github.com/projectchrono/chrono/blob/b22922cadcc0bc1271290103cb1a4448df5dd74e/src/chrono_fea/ChGaussPoint.cpp#L33 "Destructor for ChGaussPoint")
* [ChLinkMasked](https://github.com/projectchrono/chrono/blob/930606bde27a1c25bec371f7b73fb88f112d47c6/src/chrono/physics/ChLinkMasked.cpp#L73 "Destructor for ChLinkMasked")

**Comment by rserban** [comment]
Agreed: these can and should be removed.  We do this type of cleanup whenever we get a chance and I was aware of this particular one.  Having said that, it would be fantastic if you could submit a PR with this particular fix. Thanks for bringing it up.

**Comment by uebian** [comment]
They are already removed.

**Comment by elfring** [comment]
:eyes: Corresponding source code adjustments were performed with [the commit “Do not initialize (to zero) a ChMatrixDynamic on construction or on resizing”](https://github.com/projectchrono/chrono/commit/be15b0eef141fa115460be55a8bfc2955e3f99aa#diff-2882c7a83bc8afc60a1b18c505a68d331627ef83d0217230e1fc3c82f9567801L34-R26 "Adjustments from 2019-05-02").

**Comment by elfring** [comment]
:eyes: Further update candidates were left over:
* https://github.com/projectchrono/chrono/blob/d57291c61f2ed0cac04a41c50ff7fdff7db06b9a/src/chrono/physics/ChConveyor.cpp#L57-L64
* https://github.com/projectchrono/chrono/blob/d57291c61f2ed0cac04a41c50ff7fdff7db06b9a/src/chrono/physics/ChLinkMask.cpp#L42-L44
* https://github.com/projectchrono/chrono/blob/d57291c61f2ed0cac04a41c50ff7fdff7db06b9a/src/chrono/utils/ChSocketCommunication.cpp#L35-L42

:thought_balloon: Would you become interested to [use a development tool like “clang-tidy” for corresponding source code adjustments](https://clang.llvm.org/extra/clang-tidy/checks/readability/delete-null-pointer.html "Deletion of unnecessary null pointer checks")?

**Comment by dariofusai93** [comment]
Fixed in commit 85753726c2522b66986aa9ad41ae29778e06e722

**Comment by elfring** [comment]
Thanks for a few further source code improvements.
